File: scala/src/test/scala/Fibonacci/FibonacciSeriesTest.scala
package fibonacci

import org.scalatest.FunSuite

class FibonacciSeriesTest extends FunSuite {
  val fibonacciSequence = new FibonacciSeries()

  test("Canary"){
    assert(true)
  }

  test("test for N equals 0") {
    assert(fibonacciSequence.fibonacciRecursive(0) == List(1))
  }

  test("test for N equals 1") {
  assert(fibonacciSequence.fibonacciRecursive(1) == List(1, 1))
  }

  test("test for N equals 2") {
    assert(fibonacciSequence.fibonacciRecursive(2) == List(1, 1, 2))
  }

  test("test for N equals 5") {
    assert(fibonacciSequence.fibonacciRecursive(5) == List(1, 1, 2, 3, 5, 8))
  }

  test("test for N equals 8") {
    assert(fibonacciSequence.fibonacciRecursive(8) == List(1, 1, 2, 3, 5, 8, 13, 21, 34 ))
  }

  test("test for N equals 13") {
    assert(fibonacciSequence.fibonacciRecursive(13) == List(1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377))
  }

  test("tailRecursive test for N equals 0") {
    assert(fibonacciSequence.fibonacciTailRecursive(0) == List(1))
  }

  test("tailRecursive test for N equals 1") {
    assert(fibonacciSequence.fibonacciTailRecursive(1) == List(1, 1))
  }

  test("tailRecursive test for N equals 2") {
    assert(fibonacciSequence.fibonacciTailRecursive(2) == List(1, 1, 2))
  }

  test("tailRecursive test for N equals 5") {
    assert(fibonacciSequence.fibonacciTailRecursive(5) == List(1, 1, 2, 3, 5, 8))
  }

  test("tailRecursive test for N equals 8") {
    assert(fibonacciSequence.fibonacciTailRecursive(8) == List(1, 1, 2, 3, 5, 8, 13, 21, 34 ))
  }

  test("tailRecursive test for N equals 13") {
    assert(fibonacciSequence.fibonacciTailRecursive(13) == List(1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377))
  }
}

File: clojure/test/clojure_fibonacci/FibonacciSeries_test.clj
(ns clojure-fibonacci.FibonacciSeries-test
  (:require [expectations :refer :all]
            [clojure-fibonacci.FibonacciSeries :refer :all]))

;; Canary test
(expect true true)

;;test recursive for N equals 0
(expect [1] (fibonacciRecursive 0))

;;test recursive for N equals 1
(expect [1 1] (fibonacciRecursive 1))

;;test recursive for N equals 2
(expect [1 1 2] (fibonacciRecursive 2))

;;test recursive for N equals 5
(expect [1 1 2 3 5 8] (fibonacciRecursive 5))

;;test recursive for N equals 8
(expect [1 1 2 3 5 8 13 21 34] (fibonacciRecursive 8))

;;test recursive for N equals 13
(expect [1 1 2 3 5 8 13 21 34 55 89 144 233 377] (fibonacciRecursive 13))

;;test Tailrecursive for N equals 0
(expect [1] (fibonacciTailRecursive 0))

;;test Tailrecursive for N equals 1
(expect [1 1] (fibonacciTailRecursive 1))

;;test Tailrecursive for N equals 2
(expect [1 1 2] (fibonacciTailRecursive 2))

;;test Tailrecursive for N equals 5
(expect [1 1 2 3 5 8] (fibonacciTailRecursive 5))

;;test Tailrecursive for N equals 8
(expect [1 1 2 3 5 8 13 21 34] (fibonacciTailRecursive 8))

;;test Tailrecursive for N equals 13
(expect [1 1 2 3 5 8 13 21 34 55 89 144 233 377] (fibonacciTailRecursive 13))
